healthcheck/utils.go

```diff
@@ -5,18 +5,21 @@ import (
 	"time"
 )
 
-const ()
+const jitterFactor = 0.05
 
-// calcIntervalWithJitter returns a new duration based on a provided interval and a jitter percentage of 5%
+// init seeds rand at app startup
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+// calcIntervalWithJitter returns a new duration based on a provided interval and a jitter of ±jitterFactor
 func calcIntervalWithJitter(interval time.Duration) time.Duration {
-	const jitterAmount = 0.05
-	upperJitterThreshold := int(float64(interval) * jitterAmount)
-	lowerJitterThreshold := -1 * upperJitterThreshold
-	jitterToApply := time.Duration(random(lowerJitterThreshold, upperJitterThreshold))
+	upperJitterThreshold := int64(float64(interval) * jitterFactor)
```

Comment:
I think it is better to be explicit with defining both lower and upper jitter thresholds, but would have a called it maxJitter and minJitter.

like so:

```
fun getIntervalWithJitter(interval time.Duration) time.Duration {
  maxJitter := int(float64(interval) * jitterAmount)
  minJitter := -maxJitter
  jitter := time.Duration(random(minJitter, maxJitter)

  return interval + jitter
}
```

Reply:
you and Dan out-vote me on the clarity of two vars, so I'll concede this 👍